### PR TITLE
[add] run s2p on remote images (/vsicurl/http://..)

### PR DESCRIPTION
**Pull request by gfacciol**
This PR adds support for reading the RPC from remote image through vsicurl. 
Now we can seamlessly run s2p on images from internet.

### CODE REVIEW

File: s2plib/initialization.py [hunk 1]
```diff
@@ -48,10 +48,17 @@ def check_parameters(d):
...
+        img = d['images'][i]
+        if not dict_has_keys(img, ['img']):
+            print('ERROR: missing img paths for image', img)
             sys.exit(1)
+        if not dict_has_keys(img, ['rpc']) or img['rpc'] == '':
+            import tempfile     # TODO: fix common.tmpfile failure
+            img['rpc'] = tempfile.mktemp('.rpc')
+            rpc_utils.rpc_from_geotiff(img['img'], img['rpc'])
+            print('INFO: trying reading rpc from image', img)
+    print (d)
```
Comment (jmichel-otb) [inline s2plib/initialization.py]: Is this print mandatory or it is a trace left for debuging ?

File: s2plib/rpc_utils.py [hunk 2]
```diff
@@ -704,3 +704,38 @@ def crop_corresponding_areas(out_dir, images, roi, zoom=1):
...
+        path = '/vsicurl/{}'.format(geotiff_path)
+    else:
+        path = geotiff_path
+
+    f = open(outrpcfile, 'wb')
+    x = subprocess.Popen(["gdalinfo", path], stdout=subprocess.PIPE).communicate()[0]
+    x = x.splitlines()
+    for l in x:
+
+        if(1):
```
Comment (jmichel-otb) [inline s2plib/rpc_utils.py]: if(1) always true

**Review by jmichel-otb — CHANGES_REQUESTED** [review]
Two minor comments

File: s2plib/initialization.py (hunk 1 above)
Comment (gfacciol) [reply s2plib/initialization.py]: it was a debug leftover, I'm fixing it.

**Review by jmichel-otb — APPROVED** [review]

**Pull request merged**